provider/pkg/gen/schema.go

```diff
@@ -519,6 +524,30 @@ func (g *packageGenerator) genResource(typeName string, dd discoveryDocumentReso
 	return nil
 }
 
+func (g *packageGenerator) genResourceOverlays(typeName string, dd discoveryDocumentResource, patternParams codegen.StringSet) error {
+	resourceTok := g.genToken(typeName)
+	switch resourceTok {
+	case "google-native:container/v1:Cluster", "google-native:container/v1beta1:Cluster":
+		resourceMeta := g.metadata.Resources[resourceTok]
+		updateRequest := g.rest.Schemas[dd.updateMethod.Request.Ref]
+		updateBag, err := g.genProperties(typeName, &updateRequest, "update", false, patternParams)
+		if err != nil {
+			return err
+		}
+		for name, value := range updateBag.properties {
+			if _, has := resourceMeta.CreateProperties[stripDesiredPrefix(name)]; has {
+				value.SdkName = stripDesiredPrefix(name)
+				resourceMeta.UpdateProperties[name] = value
+			} else if name == "desiredClusterAutoscaling" {
+				value.SdkName = "autoscaling"
+				resourceMeta.UpdateProperties[name] = value
+			}
```

Comment:
Which properties would fall into a final `else` statement here? Should we do anything about them?